packages/neo-one-client-core/src/user/DapiUserAccountProvider.ts

```diff
@@ -0,0 +1,394 @@
+import {
+  AddressString,
+  addressToScriptHash,
+  Attribute,
+  ClaimTransaction,
+  common,
+  ContractTransaction,
+  Input,
+  InvalidParamError,
+  InvocationTransaction,
+  NetworkType,
+  Output,
+  PublicKeyString,
+  ScriptBuilderParam,
+  TransactionReceipt,
+  TransactionResult,
+  Transfer,
+  UserAccount,
+  UserAccountID,
+  UserAccountProvider,
+} from '@neo-one/client-common';
+import BigNumber from 'bignumber.js';
+import { BN } from 'bn.js';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { distinctUntilChanged } from 'rxjs/operators';
+import { InvocationCallError, NothingToTransferError, NotImplementedError } from '../errors';
+import {
+  ACCOUNT_CHANGED,
+  Argument,
+  AssetInput,
+  AssetOutput,
+  Dapi,
+  DapiAccount,
+  DapiNetworks,
+  NETWORK_CHANGED,
+  TxHashAttribute,
+} from './Dapi';
+import {
+  ExecuteInvokeClaimOptions,
+  ExecuteInvokeMethodOptions,
+  ExecuteInvokeScriptOptions,
+  Provider,
+  UserAccountProviderBase,
+} from './UserAccountProviderBase';
+
+// Maintain consistency with @neo-one/client-common/src/ScriptBuilder.ts
+const paramToArgDataType = (value: ScriptBuilderParam | undefined): Argument => {
```

Comment:
I wonder if there's a way that we can consolidate the logic between here and `ScriptBuilder` as opposed to the comment, which may be easily missed if we change something small in the function (here or there).